deployer/deployer-services/src/main/java/io/hyscale/deployer/services/deployer/impl/KubernetesDeployer.java

```diff
@@ -169,6 +192,36 @@ public boolean authenticate(AuthConfig authConfig) {
         // TODO
         return false;
     }
+    
+    
+	@Override
+	public void exec(K8sAuthorisation k8sAuthorisation, String servicename, String appname, String namespace,
+			String replicaName) {
+
+		List<String> commands = null;
+		KubeConfig kubeConfig = null;
+		if (k8sAuthorisation != null && k8sAuthorisation.getK8sAuthType() == K8sAuthType.BASIC_AUTH) {
+			K8sBasicAuth k8sBasicAuth = (K8sBasicAuth) k8sAuthorisation;
+			commands = ExecCommandUtil.getExecCommand(replicaName, namespace, k8sBasicAuth);
+		} else if (k8sAuthorisation != null && k8sAuthorisation.getK8sAuthType() == K8sAuthType.KUBE_CONFIG_FILE) {
+			K8sConfigFileAuth k8sConfigFileAuth = (K8sConfigFileAuth) k8sAuthorisation;
+			kubeConfig = getKubeConfig(k8sConfigFileAuth);
+			if (kubeConfig.getAccessToken() != null) {
+				commands = ExecCommandUtil.getExecCommand(replicaName, namespace, kubeConfig.getAccessToken());
+				CommandExecutor.executeExec(commands);
+			}
+		}
+	}
```

Comment:
Extract out the null check and throw exception if k8sAuthorisation is null
Then should we use switch with k8sAuthorisation.getK8sAuthType()? As later we may have more cases to support.
Should we have the logic to get commands in DeployCommandProvider on the lines of ImageCommandProvider
It would take k8sAuthorisation and other variable and return the command. No need to have that logic here.